lib/widgets/widget_list_item.dart

```diff
@@ -0,0 +1,143 @@
+import 'package:cogboardmobileapp/constants/constants.dart';
+import 'package:cogboardmobileapp/models/dashboard_tab_model.dart';
+import 'package:cogboardmobileapp/models/widget_model.dart';
+import 'package:cogboardmobileapp/providers/filter_provider.dart';
+import 'package:cogboardmobileapp/screens/widget_screen.dart';
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+class WidgetListItem extends StatelessWidget {
+  final DashboardWidget widget;
+  final int widgetIndex;
+  final DashboardType dashboardType;
+
+  WidgetListItem({@required this.widget, @required this.widgetIndex, @required this.dashboardType});
+
+  @override
+  Widget build(BuildContext context) {
+    return Card(
+      color: getWidgetColor(widget),
+      elevation: 5,
+      margin: widgetIndex == 0 ? EdgeInsets.fromLTRB(16, 0, 16, 8) : EdgeInsets.symmetric(
+        vertical: 8,
+        horizontal: 16,
+      ),
+      child: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: ListTile(
+          title: Text(
+            widget.title,
+            style: TextStyle(color: Colors.white),
+          ),
+          trailing: getWidgetIcon(widget),
+          onTap: () {
+            Navigator.pushNamed(
+              context,
+              DashboardItemScreen.routeName,
+              arguments: widget,
+            ).then((_) => Provider.of<FilterProvider>(context, listen: false).resetFilterView());
+          },
+        ),
+      ),
+    );
+  }
+
+  Color getWidgetColor(DashboardWidget dashboardWidget) {
+    if (dashboardWidget.content.containsKey('widgetStatus')) {
+      switch (dashboardWidget.content['widgetStatus']) {
```

Comment:
If I understand this correctly, we could use "if-else" statement instead of "switch-case", for example:
`dynamic widgetStatus = dashboardWidget.content['widgetStatus'];`
     
     if (widgetStatus is WidgetStatus) {
        return StatusColors[widgetStatus];
      } else {
        return StatusColors["DEFAULT"];
      }

This will make code shorter, easier to read and we will not need to remember to add additional "case" statement when new `WidgetStatus` field would be created.

Reply:
We could wrap whole method to shorter form:

``` 
  
const WIDGET_STATUS_KEY = 'widgetStatus';

Color getWidgetColor(DashboardWidget dashboardWidget) {
    if (dashboardWidget.content.containsKey(WIDGET_STATUS_KEY)) {
      dynamic widgetStatus = dashboardWidget.content[WIDGET_STATUS_KEY];
      if (widgetStatus is WidgetStatus) return StatusColors[widgetStatus];
    }
    return StatusColors["DEFAULT"];
  }
```

`const WIDGET_STATUS_KEY = 'widgetStatus';` can be defined in example in widget_model.dart file - this will prevent us from changing 'widgetStatus' key in every place in code if key would change

Reply:
good point, adjustments added